### PR TITLE
[SPARK-18871][SQL][TESTS] New test cases for IN/NOT IN subquery 2nd batch

### PR DESCRIPTION
**Pull request by kevinyu98**
## What changes were proposed in this pull request?

This is 2nd batch of test case for IN/NOT IN subquery.  In this PR, it has these test cases:
`in-limit.sql`
`in-order-by.sql`
`not-in-group-by.sql`

These are the queries and results from running on DB2.
[in-limit DB2 version](https://github.com/apache/spark/files/743267/in-limit.sql.db2.out.txt)
[in-order-by DB2 version](https://github.com/apache/spark/files/743269/in-order-by.sql.db2.txt)
[not-in-group-by DB2 version](https://github.com/apache/spark/files/743271/not-in-group-by.sql.db2.txt)
[output of in-limit.sql DB2](https://github.com/apache/spark/files/743276/in-limit.sql.db2.out.txt)
[output of in-order-by.sql DB2](https://github.com/apache/spark/files/743278/in-order-by.sql.db2.out.txt)
[output of not-in-group-by.sql DB2](https://github.com/apache/spark/files/743279/not-in-group-by.sql.db2.out.txt)

## How was this patch tested?

This pr is adding new test cases.

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/not-in-group-by.sql.out [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                                  FROM   t3
+                                  WHERE  t3a = t2a
+                                  GROUP  BY t3b) order by t2a)
+-- !query 7 schema
+struct<t1a:string,t1b:smallint>
+-- !query 7 output
+val1a	16
+val1a	16
+val1a	6
+val1a	6
```
Comment (dilipbiswal) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/not-in-group-by.sql.out]: I have compared the result set matched with the result from DB2.

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-order-by.sql.out [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+ORDER  BY t1b DESC nulls last
+-- !query 17 schema
+struct<count(DISTINCT t1a):bigint,t1b:smallint>
+-- !query 17 output
+1	16
+1	10
+1	10
+1	8
+1	6
+1	NULL
```
Comment (dilipbiswal) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-order-by.sql.out]: I have compared the result set matched with the result from DB2.

File: sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-limit.sql.out [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                   FROM   t2
+                   ORDER  BY t2b DESC nulls first
+                   LIMIT 1)
+GROUP  BY t1b
+ORDER BY t1b NULLS last
+LIMIT  1
+-- !query 7 schema
+struct<count(DISTINCT t1a):bigint,t1b:smallint>
+-- !query 7 output
+1	6
```
Comment (dilipbiswal) [inline sql/core/src/test/resources/sql-tests/results/subquery/in-subquery/in-limit.sql.out]: I have compared the result set matched with the result from DB2.

**Comment by hvanhovell** [comment]
LGTM - merging to master

**Pull request closed without merging**